lib/core-integration/src/Test/Integration/Scenario/API/Shelley/SharedWallets.hs

```diff
@@ -0,0 +1,522 @@
+        let wal@(ApiSharedWallet (Right _activeWal)) = getFromResponse id rPost
+
+        let payloadPatch = Json [json| {
+                "account_public_key": #{accXPub1},
+                "cosigner": "cosigner#1"
+                } |]
+
+        rPatch <- patchSharedWallet ctx wal Payment payloadPatch
+        expectResponseCode HTTP.status403 rPatch
+        expectErrorMessage errMsg403NotPendingWallet rPatch
+
+    it "SHARED_WALLETS_PATCH_04 - Cannot add cosigner key when delegation script missing and cannot add already existant key to other cosigner" $ \ctx -> runResourceT $ do
+        let payloadCreate = Json [json| {
+                "name": "Shared Wallet",
+                "account_public_key": #{accXPubTxt0},
+                "account_index": "30H",
+                "payment_script_template":
+                    { "cosigners":
+                        { "cosigner#0": #{accXPubTxt0} },
+                      "template":
+                          { "all":
+                             [ "cosigner#0",
+                               "cosigner#1",
+                               { "active_from": 120 }
+                             ]
+                          }
+                    }
+                } |]
+        rPost <- postSharedWallet ctx Default payloadCreate
+        expectResponseCode HTTP.status201 rPost
+        let wal@(ApiSharedWallet (Left _pendingWal)) = getFromResponse id rPost
+
+        let payloadPatch = Json [json| {
+                "account_public_key": #{accXPub1},
+                "cosigner": "cosigner#1"
+                } |]
+
+        rPatch <- patchSharedWallet ctx wal Delegation payloadPatch
+        expectResponseCode HTTP.status403 rPatch
+        expectErrorMessage errMsg403NoDelegationTemplate rPatch
+
+    it "SHARED_WALLETS_PATCH_05 - Can update key of cosigner in pending state" $ \ctx -> runResourceT $ do
+        let payloadCreate = Json [json| {
+                "name": "Shared Wallet",
+                "account_public_key": #{accXPubTxt0},
+                "account_index": "30H",
+                "payment_script_template":
+                    { "cosigners":
+                        { "cosigner#0": #{accXPubTxt0} },
+                      "template":
+                          { "all":
+                             [ "cosigner#0",
+                               "cosigner#1",
+                               { "active_from": 120 }
+                             ]
+                          }
+                    }
+                } |]
+        rPost <- postSharedWallet ctx Default payloadCreate
+        expectResponseCode HTTP.status201 rPost
+        let wal@(ApiSharedWallet (Left _pendingWal)) = getFromResponse id rPost
+
+        let payloadPatch = Json [json| {
+                "account_public_key": #{accXPub1},
+                "cosigner": "cosigner#0"
+                } |]
+
+        rPatch <- patchSharedWallet ctx wal Payment payloadPatch
+        expectResponseCode HTTP.status200 rPatch
```

Comment:
I would actually not expect this to work. Updating "our" key shouldn't be possible as this would lead to a different identifier (and therefore, a different shared wallet from the API).

Reply:
I thought we enable updating of keys for a given cosigner but only when the shared wallet is in pending state. When in active state it is not longer possible. So we should outlaw it even in pending state (in the not complete process of collecting keys for all cosigners) ?

Reply:
yes, I see. updating should be possible for other cosigners in pending state. not to ourselves. agree

Reply:
it is fixed in a sense that when we have different key than shared wallet key, we can update to it (when we are in pending state). But when we have cosigner with wallet shared key we cannot update it  as shown in PATCH_07 test -> https://github.com/input-output-hk/cardano-wallet/pull/2549#discussion_r613857829 Here I needed to fix this test to be able to patch it successfully